src/libcore/mem.rs

```diff
@@ -611,3 +611,80 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
+/// representation is. Note that if the enum's representation is signed, then
+/// the discriminant values returned by this function will not compare in the
+/// same order as actual enum variants would under `#[derive(PartialOrd)]`.
+///
+/// If `T` is not an enum, the return value is unspecified (this is **not** undefined
+/// behavior, but the defined behavior is that the value is unpredictable and subject
+/// to change).
```

Comment:
Feels like not a proper place to explain what a difference between unspecified value and undefined behaviour is. I feel like the most of the text between parenthesis is not necessary and the paragraph could instead read just:

> If `T` is not an enum, the return value is unspecified, but does not result in undefined behaviour.